src/vm/dispparammarshaler.cpp

```diff
@@ -251,11 +251,19 @@ void DispParamArrayMarshaler::MarshalNativeToManaged(VARIANT *pSrcVar, OBJECTREF
         if (!pElemMT && vt == VT_RECORD)
             pElemMT = OleVariant::GetElementTypeForRecordSafeArray(pSafeArray).GetMethodTable();
 
+        MethodDesc* pStructMarshalStub = nullptr;
+        if ((vt == VT_RECORD || vt == VTHACK_NONBLITTABLERECORD) && (pElemMT != nullptr && !pElemMT->IsBlittable()))
+        {
+            GCX_PREEMP();
```

Comment:
Since we are entering Preemptive mode, can't `pDestObj` be moved or is that an out? If this is an out then `*pDestObj` should be `nullptr` and I would like to see that asserted in the contract.